### DISCUSSION TITLE
Is MFA / Multi Factor Authentication supported?

### DISCUSSION THREAD

**Discussion by tam5** [beekeeper-studio/beekeeper-studio#1022] [Q&A]
My company requires a mfa token to be entered when connecting via ssh. I've seen this supported in things like sequel pro and table plus. I'm wondering if this is supported in Beekeeper or if there are any plans to add it?

**Comment by dhorrall** [comment]
I have this same question.  Doesn't appear to be, at least in Community edition.  Is it supported in paid edition?

**Reply by davidkaufman** [reply]
Looking for more information on this request. Are your company's MFA tokens generated by a Google Authenticator-compatible app, as described here? https://ubuntu.com/tutorials/configure-ssh-2fa#2-installing-and-configuring-required-packages

Also (not sure if it's relevant but) what platform are you running Beekeeper Studio on? Windows, Mac and/or Linux?

How does that work when you log in to ssh from the command line?  Does your ssh binary get some info from the server that prompts you for your token?

Thanks!

-dave

**Comment by dhorrall** [comment]
Windows...

I don't know about the MFA tokens...all I know is when I use another sql
tool, it has a choic


On Sat, Aug 12, 2023 at 3:02 PM David Kaufman ***@***.***>
wrote:

> Looking for more information on this request. Are your company's MFA
> tokens generated by a Google Authenticator-compatible app, as described
> here?
> https://ubuntu.com/tutorials/configure-ssh-2fa#2-installing-and-configuring-required-packages
>
> Also (not sure if it's real ent but) what platform are you running
> Beekeeper Studio on? Windows, Mac and/or Linux?
>
> How does that work when you log in to ssh from the command line? Does your
> ssh binary get some info from the server that prompts you for your token?
>
> Thanks!
>
> -dave
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/beekeeper-studio/beekeeper-studio/discussions/1022#discussioncomment-6710270>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFMEHIUAUCWKJESAEWFOHOLXU7HN5ANCNFSM5N4KHNSQ>
> .
> You are receiving this because you commented.Message ID:
> <beekeeper-studio/beekeeper-studio/repo-discussions/1022/comments/6710270@
> github.com>
>


-- 
Regards,
Derek Horrall

**Comment by dhorrall** [comment]
Sorry..hit send too soon...

Another SQL tool has an authentication choice like this:  I can't connect
unless I choose this. Beekeeper does not have this option, so I cannot
connect to db.

[image: image.png]



On Sat, Aug 19, 2023 at 8:55 AM Derek Horrall ***@***.***> wrote:

> Windows...
>
> I don't know about the MFA tokens...all I know is when I use another sql
> tool, it has a choic
>
>
> On Sat, Aug 12, 2023 at 3:02 PM David Kaufman ***@***.***>
> wrote:
>
>> Looking for more information on this request. Are your company's MFA
>> tokens generated by a Google Authenticator-compatible app, as described
>> here?
>> https://ubuntu.com/tutorials/configure-ssh-2fa#2-installing-and-configuring-required-packages
>>
>> Also (not sure if it's real ent but) what platform are you running
>> Beekeeper Studio on? Windows, Mac and/or Linux?
>>
>> How does that work when you log in to ssh from the command line? Does
>> your ssh binary get some info from the server that prompts you for your
>> token?
>>
>> Thanks!
>>
>> -dave
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/beekeeper-studio/beekeeper-studio/discussions/1022#discussioncomment-6710270>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AFMEHIUAUCWKJESAEWFOHOLXU7HN5ANCNFSM5N4KHNSQ>
>> .
>> You are receiving this because you commented.Message ID:
>> <beekeeper-studio/beekeeper-studio/repo-discussions/1022/comments/6710270
>> @github.com>
>>
>
>
> --
> Regards,
> Derek Horrall
>


-- 
Regards,
Derek Horrall

**Comment by davidkaufman** [comment]
Thanks @dhorrall

It looks like you tried to attach an image by replying to the GitHub notification email, but I can't see it, just a broken link.  Can you copy and paste the image into the GitHub issue via a web browser?

Also, I'm still not sure exactly what you (or @tam5 who opened this ticket originally) mean when you say an "mfa token".

Are you talking about one of the devices or apps that continually generates new, rotating time-based codes?
like the Authy App <https://authy.com/>
or Google Authenticator <https://en.wikipedia.org/wiki/Google_Authenticator>

Or a device that you plug into your phone or USB port on your computer?

Thanks!

-dave
